android/src/main/java/com/reactnativeone/OneModule.java

```diff
@@ -72,62 +82,160 @@ public void init(String siteKey, String touchpointUri, String apiKey,
+          notifyProblem(promise, Integer.toString(error.getHttpStatusCode()), error.getLocalizedMessage());
+          Log.e(NAME, "[Thunderhead] Send Properties Api Error: " + error.getErrorMessage());
+        } catch (Exception error) {
+          notifyProblem(promise, Integer.toString(error.hashCode()), error.getLocalizedMessage());
+          Log.e(NAME, "[Thunderhead] Send Interaction Error: " + error.getLocalizedMessage());
+        }
+      }
+    });
   }
 
   @ReactMethod
-  public void sendResponseCode(String responseCode, String interaction) {
-    final OneResponseCodeRequest responseCodeRequest = new OneResponseCodeRequest.Builder()
-      .responseCode(new OneResponseCode(responseCode))
-      .interactionPath(new OneInteractionPath(URI.create(interaction)))
-      .build();
-    One.sendResponseCode(responseCodeRequest).enqueue(null);
+  public void sendResponseCode(String interactionPath, String responseCode, final Promise promise) {
+    executor.submit(() -> {
+      final OneResponseCodeRequest responseCodeRequest = new OneResponseCodeRequest.Builder()
+        .responseCode(new OneResponseCode(responseCode))
+        .interactionPath(new OneInteractionPath(URI.create(interactionPath)))
+        .build();
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+        try {
+          One.sendResponseCode(THROW_ERRORS, responseCodeRequest).join();
+          notifyResult(promise, null);
+        } catch (ExecutionException error) {
+          notifyProblem(promise, Integer.toString(error.hashCode()), error.getLocalizedMessage());
+          Log.e(NAME, "[Thunderhead] Send Response Code Completion Error: " + error.getCause());
+        } catch (OneSDKError error) {
+          notifyProblem(promise, Integer.toString(error.getSystemCode()), error.getLocalizedMessage());
+          Log.e(NAME, "[Thunderhead] Send Response Code SDK Error: " + error.getErrorMessage());
+        } catch (OneAPIError error) {
+          notifyProblem(promise, Integer.toString(error.getHttpStatusCode()), error.getLocalizedMessage());
+          Log.e(NAME, "[Thunderhead] Send Response Code Api Error: " + error.getErrorMessage());
+        } catch (CompletionException error) {
+          notifyProblem(promise, Integer.toString(error.hashCode()), error.getLocalizedMessage());
+          Log.e(NAME, "[Thunderhead] Send Properties Completion Error: " + error.getLocalizedMessage());
+        } catch (Exception error) {
+          notifyProblem(promise, Integer.toString(error.hashCode()), error.getLocalizedMessage());
+          Log.e(NAME, "[Thunderhead] Send Interaction Error: " + error.getLocalizedMessage());
+        }
+      } else {
+        try {
+          One.sendResponseCodeLegacySupport(THROW_ERRORS, responseCodeRequest).join();
+          notifyResult(promise, null);
+        } catch (ExecutionException error) {
+          notifyProblem(promise, Integer.toString(error.hashCode()), error.getLocalizedMessage());
```

Comment:
You should look at the cause of the exception and send that message instead to provide better insight.